### PR TITLE
[pytorch][graviton][canary] Fix graviton canary DLC pull test

### PR DESCRIPTION
**Pull request by xncqr**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [x] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [x] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/test_utils/__init__.py [hunk 1]
```diff
@@ -833,6 +833,19 @@ def parse_canary_images(framework, region):
     """
     customer_type = get_customer_type()
     customer_type_tag = f"-{customer_type}" if customer_type else ""
+    
+    # initialize graviton variables
+    use_graviton = False
+
+    # seperating framework from regex match pattern for graviton as it is ARCH_TYPE instead of FRAMEWORK
```
Comment (shantanutrip) [inline test/test_utils/__init__.py:840]: Thanks for adding this comment

File: test/test_utils/__init__.py [hunk 2]
```diff
@@ -833,6 +833,19 @@ def parse_canary_images(framework, region):
...
+    canary_type = framework
+
+    # Setting whether Graviton Arch is used
+    # NOTE: If Graviton arch is used with a framework, not in the list below, the match search will "KeyError".
+    if os.getenv("ARCH_TYPE") == "graviton":
+        use_graviton = True
+        graviton_tag = "-graviton"
```
Comment (shantanutrip) [inline test/test_utils/__init__.py:847]: nit: Are we using this variable? If not, lets remove this?

File: test/test_utils/__init__.py [hunk 3]
```diff
@@ -928,13 +946,20 @@ def parse_canary_images(framework, region):
                     f"{registry}.dkr.ecr.{region}.amazonaws.com/autogluon-training:{fw_version}-gpu-{py_version}",
                     f"{registry}.dkr.ecr.{region}.amazonaws.com/autogluon-training:{fw_version}-cpu-{py_version}",
                 ],
+                "graviton_tensorflow": [
+                    f"{registry}.dkr.ecr.{region}.amazonaws.com/tensorflow-inference-graviton:{fw_version}-cpu-{py_version}",
+                ],
+                "graviton_pytorch": [
+                    f"{registry}.dkr.ecr.{region}.amazonaws.com/pytorch-inference-graviton:{fw_version}-cpu-{py_version}",
+                ],
```
Comment (shantanutrip) [inline test/test_utils/__init__.py:954]: nit: Lets make a TODO comment to add graviton_mxnet in the future?

File: test/test_utils/__init__.py (hunk 2 above)
Comment (xncqr) [reply test/test_utils/__init__.py:847]: We are still using it at line 898.

File: test/test_utils/__init__.py (hunk 3 above)
Comment (xncqr) [reply test/test_utils/__init__.py:954]: sounds good. adding

**Comment by xncqr** [comment]
This PR is to solve the following which passes here:
JOB: dlc-pr-pytorch-graviton_dlc-pr-sanity-test-graviton
- PASSED sanity/test_canary_integration.py::test_canary_images_pullable

The other failures in the above job are not related to this PR:
FAILED sanity/test_safety_report_file.py::test_safety_file_exists_and_is_valid
FAILED sanity/test_anaconda.py::test_repo_anaconda_not_present
FAILED sanity/test_pre_release.py::test_pip_check8-2022-05-23-21-42-14
FAILED sanity/test_pre_release.py::test_dependency_check_graviton_cpu

the failure of the following jobs are will be resolved in a future PR:
dlc-pr-pytorch-graviton_dlc-pr-ec2-test
dlc-pr-pytorch-graviton_dlc-pr-ecs-test
dlc-pr-pytorch-graviton_dlc-pr-eks-test

**Review by shantanutrip — APPROVED** [review]
LGTM. Please revert the temp changes and merge.

**Review by shantanutrip — APPROVED** [review]
LGTM

**Pull request merged**